src/lib/pages/frontsite/PatronProfile/LoansList.js

```diff
@@ -1,37 +1,39 @@
-import { Pagination } from '@components/Pagination';
+import { RenderPagination } from './PaginationShow';
 import PropTypes from 'prop-types';
 import React, { Component } from 'react';
-import { Container, Item } from 'semantic-ui-react';
+import { Item } from 'semantic-ui-react';
 
 export default class LoansList extends Component {
+  renderPagination = () => {
+    const { activePage, isLoading, onPageChange, loans, rowsPerPage } =
+      this.props;
+
+    return (
+      <RenderPagination
+        currentPage={activePage}
+        currentSize={rowsPerPage}
+        loading={isLoading}
+        onPageChange={onPageChange}
+        items={loans}
+      />
+    );
+  };
+
   render() {
-    const {
-      activePage,
-      isLoading,
-      onPageChange,
-      loans,
-      rowsPerPage,
-      renderListEntry,
-      noLoansCmp,
-    } = this.props;
+    const { loans, renderListEntry, noLoansCmp, showAllLink } = this.props;
+
     return loans.total > 0 ? (
       <>
+        {loans.hits.length > 2 && this.renderPagination()}
+        {showAllLink && showAllLink}
         <Item.Group divided>
           {loans.hits.map((loan) => (
             <React.Fragment key={loan.metadata.pid}>
               {renderListEntry(loan)}
             </React.Fragment>
           ))}
         </Item.Group>
-        <Container textAlign="center">
-          <Pagination
-            currentPage={activePage}
-            currentSize={rowsPerPage}
-            loading={isLoading}
-            onPageChange={onPageChange}
-            totalResults={loans.total}
-          />
-        </Container>
+        {this.renderPagination()}
```

Comment:
is there any reason of creating a new method for this JSX code?

Reply:
I would say in this case is not code duplication since there is no "logic" but only component usage 
```
      <RenderPagination
        currentPage={activePage}
        currentSize={rowsPerPage}
        loading={isLoading}
        onPageChange={onPageChange}
        items={loans}
      />
```

code duplication would be more if you would perform some actions repeatedly